src/index.ts

```diff
@@ -57,12 +57,13 @@ export function createAuthMiddleware (opts: CreateAuthOptions) {
     }
 
     // Update session
-    if (authRes.session) {
+    if (authRes.authorized && authRes.session) {
```

Comment:
We would need updating session when not authorized as well in order to save/restore original URL when coming back